### PR TITLE
38 add logger detail

### PR DESCRIPTION
**Pull request by bradleySuira**
#38

### CODE REVIEW

File: frontend/src/components/Simulator/NodeLogs.vue [hunk 1]
```diff
@@ -4,13 +4,22 @@ import { webSocketClient } from '@/utils'
...
+const colorMap: Record<string, string> = {
+  'info': 'text-blue-500',
+  'error': 'text-red-500',
+  'warning': 'text-yellow-500',
+  'success': 'text-green-500'
+}
 
-
+const getDataText = (data: any) => {
+  if (typeof data !== 'string') return JSON.stringify(data)
```
Comment (cristiam86) [inline frontend/src/components/Simulator/NodeLogs.vue:15]: @bradleySuira what if we do `JSON.stringify(data, null, 2)` to print it a little bit better?

File: genvm/server.py [hunk 2]
```diff
@@ -7,35 +7,39 @@
...
 def leader_executes_transaction(contract_code: str, node_config: dict) -> dict:
 
     delete_recipts()
 
-    return_data = {"status": "error", "data": None}
+    msg = MessageHandler(app, socketio)
 
     icontract_file, _, _, leader_recipt_file = transaction_files()
 
+    msg.debug_response("icontract", contract_code)
```
Comment (cristiam86) [inline genvm/server.py:40]:
```suggestion
    msg.debug_response("Contract Code", contract_code)
```

File: genvm/server.py [hunk 3]
```diff
@@ -76,11 +80,17 @@ def validator_executes_transaction(
...
 
     delete_recipts()
+    
+    msg = MessageHandler(app, socketio)
 
     return_data = {"status": "error", "data": None}
 
     icontract_file, recipt_file, _, _ = transaction_files()
 
+    msg.debug_response("icontract", icontract)
```
Comment (cristiam86) [inline genvm/server.py:90]:
```suggestion
    msg.debug_response("Contract Code", contract_code)
```

File: rpc/utils.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import re
+import string
+import random
+
+
+def create_new_address() -> str:
```
Comment (cristiam86) [inline rpc/utils.py:6]: @sevenearths these two methods about the address look like they should be at the protocol level. @jmlago do you think they should be in the GenVM?

**Review by cristiam86 — CHANGES_REQUESTED** [review]

File: common/config/logging.dev.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "formatters": {
+        "default": {
+            "format": "[%(asctime)s] %(levelname)s | %(module)s >>> %(message)s",
+            "datefmt": "%B %d, %Y %H:%M:%S %Z"
+        },
+        "simple": {
+            "format": "%(levelname)s | %(message)s"
+        }
+    },
+    "handlers": {
```
[the remark this answers is not in the document]
Comment (sevenearths) [reply common/config/logging.dev.json:12]: Yup. Your right. I'll fix that.

File: frontend/src/components/Simulator/NodeLogs.vue (hunk 1 above)
Comment (sevenearths) [reply frontend/src/components/Simulator/NodeLogs.vue:15]: I'll add that

File: genvm/server.py (hunk 2 above)
Comment (sevenearths) [reply genvm/server.py:40]: Updated!

File: genvm/server.py (hunk 3 above)
Comment (sevenearths) [reply genvm/server.py:90]: There is no variable called `contract_code`.

**Review by sevenearths — CHANGES_REQUESTED** [review]
@bradleySuira I think the frontend should poll `get_providers_and_models` every time a user wants to edit or create a node so it gets the latest information
IMG20240506111803_BURST000_COVER

The code is missing the `get_icontract_schema_for_code` function

When `deploy_intelligent_contract` is called it's missing the class method

**Review by sevenearths — APPROVED** [review]

**Comment by cristiam86** [comment]
I fixed the get_contract_state response format to match what the frontend is expecting. @bradleySuira, please take a look since I can't ask for your review here because you are the owner of the PR.

**Review by sevenearths — APPROVED** [review]

**Review by sevenearths — APPROVED** [review]

**Review by sevenearths — APPROVED** [review]

**Review by sevenearths — APPROVED** [review]

**Review by sevenearths — APPROVED** [review]

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**